### ISSUE TITLE
Custom path matching regex

### ISSUE THREAD

**Issue by fernandospr** [airbnb/DeepLinkDispatch#93]
Currently I have seen that the regex (DeepLinkEntry.PARAM_VALUE) is "([a-zA-Z0-9_#!+%~,\\-\\.\\$]*)".
However I have deeplinks that contains other characters, for example '&'.
So they will not match.

The proposal is to allow the client to customize this regex.

Thanks.

**Comment by felipecsl** [comment]
I don't think we wanna make the regex customizable at runtime, but please feel free to submit a pull request that allows `&` on the path regex

**Comment by fernandospr** [comment]
Do you see any problems making this externally configured? Like using a meta tag in the manifest? I can make a pull request to include '&', however tomorrow maybe I will need more, I don't know maybe ¡ ¿ @ etc.

**Comment by felipecsl** [comment]
I believe this regex should be following the URI specification for the [path component](https://tools.ietf.org/html/rfc3986#page-22). While the RFC does not explicitly state which characters are allowed, the rules are not supposed to change.
The problem here is that our regex is too restrictive and should just be fixed to allow more special characters
